ambari-server/src/main/resources/Ambari-DDL-Derby-CREATE.sql

```diff
@@ -157,9 +156,9 @@ CREATE TABLE clusterservices (
   cluster_id BIGINT NOT NULL,
   service_group_id BIGINT NOT NULL,
   service_enabled INTEGER NOT NULL,
-  CONSTRAINT PK_clusterservices PRIMARY KEY (id, service_group_id, cluster_id),
-  CONSTRAINT UQ_service_id UNIQUE (id),
-  CONSTRAINT FK_clusterservices_cluster_id FOREIGN KEY (service_group_id, cluster_id) REFERENCES servicegroups (id, cluster_id));
+  CONSTRAINT PK_clusterservices PRIMARY KEY (id),
+  CONSTRAINT UK_clusterservices_id UNIQUE (id, cluster_id, service_group_id, service_name),
```

Comment:
Same as above, don't use a PK as part of a unique clause.

Reply:
Fixed.